### PR TITLE
fix(shared): Remove "sideEffects" and fix postbuild script

### PR DESCRIPTION
**Pull request by LekoArts**
## Description

In the first commit I tried to bundle `@clerk/shared` into Expo but that didn't completely work as we'd also need to bundle React or Clerk JS -- ugh :/

So this PR adds the `files` and adjusts the subpaths script.

I also removed the `sideEffects: false` since tsup/esbuild complained that this package is not side-effects free. Which is true since some packages import shared chunks.

This is a follow-up to https://github.com/clerkinc/javascript/pull/1950

## Checklist

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerkinc/clerk-docs) has been updated

## Type of change

- [x] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/backend`
- [ ] `@clerk/chrome-extension`
- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/fastify`
- [ ] `gatsby-plugin-clerk`
- [ ] `@clerk/localizations`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/remix`
- [ ] `@clerk/clerk-sdk-node`
- [x] `@clerk/shared`
- [ ] `@clerk/themes`
- [ ] `@clerk/types`
- [ ] `build/tooling/chore`

### CODE REVIEW

**Comment by LekoArts** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @LekoArts - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/backend | 0.31.4-snapshot.vc1e0279 |
| @clerk/chrome-extension | 0.4.11-snapshot.vc1e0279 |
| @clerk/clerk-js | 4.64.0-snapshot.vc1e0279 |
| @clerk/clerk-expo | 0.19.13-snapshot.vc1e0279 |
| @clerk/fastify | 0.6.18-snapshot.vc1e0279 |
| gatsby-plugin-clerk | 4.4.19-snapshot.vc1e0279 |
| @clerk/nextjs | 4.25.8-snapshot.vc1e0279 |
| @clerk/clerk-react | 4.27.1-snapshot.vc1e0279 |
| @clerk/remix | 3.1.1-snapshot.vc1e0279 |
| @clerk/clerk-sdk-node | 4.12.17-snapshot.vc1e0279 |
| @clerk/shared | 1.0.1-snapshot.vc1e0279 |

Tip: Use the snippet copy button below to quickly install the required packages.
`@clerk/backend`
```sh
npm i @clerk/backend@0.31.4-snapshot.vc1e0279 --save-exact
```
`@clerk/chrome-extension`
```sh
npm i @clerk/chrome-extension@0.4.11-snapshot.vc1e0279 --save-exact
```
`@clerk/clerk-js`
```sh
npm i @clerk/clerk-js@4.64.0-snapshot.vc1e0279 --save-exact
```
`@clerk/clerk-expo`
```sh
npm i @clerk/clerk-expo@0.19.13-snapshot.vc1e0279 --save-exact
```
`@clerk/fastify`
```sh
npm i @clerk/fastify@0.6.18-snapshot.vc1e0279 --save-exact
```
`gatsby-plugin-clerk`
```sh
npm i gatsby-plugin-clerk@4.4.19-snapshot.vc1e0279 --save-exact
```
`@clerk/nextjs`
```sh
npm i @clerk/nextjs@4.25.8-snapshot.vc1e0279 --save-exact
```
`@clerk/clerk-react`
```sh
npm i @clerk/clerk-react@4.27.1-snapshot.vc1e0279 --save-exact
```
`@clerk/remix`
```sh
npm i @clerk/remix@3.1.1-snapshot.vc1e0279 --save-exact
```
`@clerk/clerk-sdk-node`
```sh
npm i @clerk/clerk-sdk-node@4.12.17-snapshot.vc1e0279 --save-exact
```
`@clerk/shared`
```sh
npm i @clerk/shared@1.0.1-snapshot.vc1e0279 --save-exact
```

File: packages/shared/package.json [hunk 1]
```diff
@@ -2,9 +2,27 @@
...
+    "handleValueOrFn",
+    "isomorphicAtob",
+    "keys",
+    "loadScript",
+    "localStorageBroadcastChannel",
+    "poller",
+    "proxy",
+    "underscore",
+    "url",
+    "react"
```
Comment (panteliselef) [inline packages/shared/package.json:25]: ❓ @LekoArts why are these added at top level ?

Comment (tmilewski) [reply packages/shared/package.json:25]: @LekoArts I was wondering the same thing as I was running into `turbo clean` issues. 

The npm script for shared only runs `rimraf` on `./dist`.

Are we able to nest these?

Thanks!

Comment (tmilewski) [reply packages/shared/package.json:25]: Additionally, we'll need to add any directories/etc into `turbo.json` as, right now, it only lists `"dist/**"` as an output.

**Review by tmilewski — CHANGES_REQUESTED** [review]

**Review by tmilewski — DISMISSED** [review]
❓ I was wondering the same thing as @panteliselef. Are we able to nest them?

We'll need to define these additional directories in two places and a nested approach will make it more manageable.

- NPM Script: `clean` will need to account for these generated files.
- Turborepo: The generated files/dirs will also need to be listed in the `build` task `outputs`.

As-is, they both only account for `./dist`.

Thanks!

**Comment by LekoArts** [comment]
This is a follow-up to https://github.com/clerkinc/javascript/pull/1950 so the functionality is not new and Turborepo does not need to be considered here as these are only package.json files for compat reasons

**Comment by nikosdouvlis** [comment]
> ❓ I was wondering the same thing as @panteliselef. Are we able to nest them?
> 
> We'll need to define these additional directories in two places and a nested approach will make it more manageable.
> 
> * NPM Script: `clean` will need to account for these generated files.
> * Turborepo: The generated files/dirs will also need to be listed in the `build` task `outputs`.
> 
> As-is, they both only account for `./dist`.
> 
> Thanks!

We're not able to nest them as we've found that Expo does not respect the `exports` field. Please see https://github.com/clerkinc/javascript/pull/1950 for more details.

File: packages/shared/package.json (hunk 1 above)
Comment (nikosdouvlis) [reply packages/shared/package.json:25]: @panteliselef Please see https://github.com/clerkinc/javascript/pull/1950

File: packages/shared/package.json [hunk 2]
```diff
@@ -2,9 +2,27 @@
   "name": "@clerk/shared",
   "version": "1.0.0",
   "description": "Internal package utils used by the Clerk SDKs",
-  "sideEffects": false,
```
Comment (nikosdouvlis) [inline packages/shared/package.json:5]: Did you notice any side-effects (pun intended) in regards to code pruning when removing this?

**Review by nikosdouvlis — APPROVED** [review]

File: packages/shared/package.json (hunk 2 above)
Comment (LekoArts) [reply packages/shared/package.json:5]: Still all worked fine during bundling :)
